### PR TITLE
Describe the behavior when using shallow clone

### PR DESCRIPTION
**Pull request by mfojtik**
Follow up for https://github.com/openshift/origin/pull/7521

### CODE REVIEW

**Comment by mfojtik** [comment]
@bparees @adellape PTAL

File: dev_guide/builds.adoc [hunk 1]
```diff
@@ -598,6 +598,16 @@ containing a Dockerfile that overwrites any Dockerfile that may exist in the
...
 
+When using the Git repository as a source without specifying the `*ref*`
+field, OpenShift will perform a shallow clone (`--depth=1` clone). That means
+only the `HEAD` (usually the `master` branch) will be downloaded. This result
+into much faster download times of repositories with rich history of commits.
+
+The shallow clone is also used when the `*ref*` field is specified and set to an
+existing remote branch name. However, if you specify the `*ref*` field to a commit
+SHA256 we have to fallback into a regular git clone operation and then checkout the
+commit, because using the `--depth=1` option only works with named branch refs.
```
Comment (bparees) [inline dev_guide/builds.adoc]: s/we have to fallback into/the system will fallback to a regular..../

also add a sentence about how you can specify "master" as your ref to force a deep clone if your build logic needs access to the commit history.

Comment (mfojtik) [inline dev_guide/builds.adoc]: fixed. thanks

File: dev_guide/builds.adoc [hunk 2]
```diff
@@ -598,6 +598,19 @@ containing a Dockerfile that overwrites any Dockerfile that may exist in the
...
 
+When using the Git repository as a source without specifying the `*ref*`
+field, OpenShift will perform a shallow clone (`--depth=1` clone). That means
+only the `HEAD` (usually the `master` branch) will be downloaded. This result
+into much faster download times of repositories with rich history of commits.
+
+The shallow clone is also used when the `*ref*` field is specified and set to an
+existing remote branch name. However, if you specify the `*ref*` field to a commit
+SHA256 the system will fallback to a regular git clone operation and then checkout the
+commit, because using the `--depth=1` option only works with named branch refs.
```
Comment (bparees) [inline dev_guide/builds.adoc]: comma not needed.

**Comment by bparees** [comment]
final nit and lgtm.  @openshift/team-documentation ptal.  Feature has merged so this can merge as soon as @mfojtik removes the extra comma.

**Comment by bfallonf** [comment]
@adellape Is there a reason this was milestoned Future Release and not Next Release?

**Comment by adellape** [comment]
@bfallonf I'm guessing because I didn't see the origin PR's commit in the ose repo (https://github.com/openshift/ose), and from pulling the latest for ose, I still don't see it in yet.

**Comment by bfallonf** [comment]
@adellape Hmm OK. I've labelled the followup PRs as future as well.

**Pull request merged**